src/light.rs

```diff
@@ -44,6 +50,24 @@ impl Ambient {
     pub(crate) fn new(object: Base) -> Self {
         Ambient { object }
     }
+
+    /// Change light color.
+    pub fn set_color(
```

Comment:
sounds like we could have a `Light` trait with those methods? Maybe even have the implementation shared somehow

Reply:
I have some concerns. 
We can use the trait and implement methods in it, then every time we want to use these methods, we should import the trait, it's annoying.
On the other hand, the implemntation with a macro is unfriendly for IDEs.
Do you prefer the case with a trait?